examples/api-samples/src/browser/sample-dynamic-label-provider-contribution.ts

```diff
@@ -58,7 +58,7 @@ export class SampleDynamicLabelProviderContribution extends DefaultUriLabelProvi
         return new URI(element.toString());
     }
 
-    async getIcon(element: URI): Promise<string> {
+    getIcon(element: URI): string {
```

Comment:
- [ ] (breaking) document that `LabelProviderContribution.getIcon` should be fast and async to avoid blocking rendering and remove icon caching from (tree) nodes